packages/ui-extensions/src/types/Button.ts

```diff
@@ -0,0 +1,58 @@
+export interface BaseButtonProps {
+  /**
+   * Destination to link to, renders a Link
+   */
+  to?: string;
+  /**
+   * The type of button that will be rendered.
+   *
+   * `primary`: button used for main actions or green-path. Ex: Continue to next step, Discount field
+   * `secondary`: button used for secondary actions not blocking user progress. Ex: Download Shop app
+   * `plain`: Renders a button that visually looks like a Link
+   * @default 'primary'*/
+  kind?: 'primary' | 'secondary' | 'plain';
+  /**
+   *  Specifies the color of the Button. The button will keep the style of the chodsen `kind`,
+   *  but replace its color according to the appearance.
+   *
+   * `monochrome`: button will inherit the color of its parent
+   * `critical`: button will take inherit the color of the Critical color group (in Checkout)
+   *  		 and map to 'destructive' (in Admin/Polaris). Typically used for destructive actions.
+   *
+   *  Excludes 'success' | 'warning' since buttons in a Banner will be `monochrome`
+   *  and use the default colors set by the Banner.
+   */
+  appearance?: 'monochrome' | 'critical';
```

Comment:
IMO this is also an admin design system prop, I also find it hard to understand the difference between this and `kind` (neither prop seems to imply it adds semantic meaning, so they all just seem like visual variations)

Reply:
`kind` in checkout defines the semantic of the button. It does come with some default visual styles, but they are editable by the merchant.

`appearance` is the visual prop. It is combined with the kind prop to offer more flexibility.

This is what is possible with the defaults styles of Checkout.
![image](https://user-images.githubusercontent.com/739905/121959179-6dda1600-cd32-11eb-822e-574ce3bb43ac.png)